### PR TITLE
feat(llm): add cycling system prompts and SystemPromptLoader (#300)

### LINKED ISSUE #300: System prompts cyclotourisme versionnés (LLaMA 8B)
## Contexte

Les system prompts pour LLaMA 8B doivent être spécialisés cyclotourisme et versionnés dans le repo. Pas de fine-tuning — tout passe par du prompt engineering vanilla avec few-shot examples.

### Prompt passe 1 — Analyse par étape (`stage-analysis.txt`)
- Rôle : analyste cyclotourisme expert
- Terminologie vélo : types de surface (asphalte, gravel, chemin), profils cyclistes, VAE
- Contexte géographique : France, Belgique, Pays-Bas (EuroVelo, voies vertes)
- Input structuré : JSON compact (~200-300 tokens) avec distance, dénivelé, surface, météo, alertes, points d'eau, ravitaillement, hébergements, heure d'arrivée, coucher de soleil
- Output : synthèse narrative + insights complémentaires + suggestions actionnables
- Few-shot : 1 exemple complet d'input → output attendu
- Contrainte : output < 500 tokens

### Prompt passe 2 — Vue d'ensemble (`trip-overview.txt`)
- Rôle : coach cyclotourisme
- Input : résumés passe 1 de toutes les étapes + profil utilisateur (~2000-3000 tokens)
- Output : fatigue cumulative, patterns cross-étapes, cohérence globale, recommandations
- Few-shot : 1 exemple de trip 3 étapes → analyse globale
- Contrainte : output < 800 tokens

### Service `SystemPromptLoader`
- Charge les fichiers `.txt` depuis `api/src/Llm/SystemPrompt/`
- Injecte les variables dynamiques (profil utilisateur, date, langue)
- Template Twig ou simple `str_replace` pour les placeholders

### Tests
- `SystemPromptLoaderTest.php` : chargement, substitution variables, fichier manquant
- Validation : vérifier que les prompts respectent les limites de tokens

## Fichiers
- Nouveau : `api/src/Llm/SystemPrompt/stage-analysis.txt`
- Nouveau : `api/src/Llm/SystemPrompt/trip-overview.txt`
- Nouveau : `api/src/Llm/SystemPromptLoader.php`
- Nouveau : `api/tests/Unit/Llm/SystemPromptLoaderTest.php`

## Effort : S

### PR DESCRIPTION
**Pull request by vincentchalamon**
Closes #300.

## Summary

- Adds two versioned LLaMA 8B system prompts under `api/src/Llm/SystemPrompt/` :
  - `stage-analysis.txt` — pass 1, per-stage analysis, role: cycling-tour analyst, output Markdown 3 sections (Synthèse / Insights / Recommandations), 1 few-shot example (Florac → Le Pont-de-Montvert), &lt;500 tokens.
  - `trip-overview.txt` — pass 2, trip overview, role: cycling-tour coach, output 4 sections (Vue d'ensemble / Charge / Patterns / Recommandations globales), few-shot 3 stages, &lt;800 tokens.
- Adds `App\Llm\SystemPromptLoader` (`final readonly`) with placeholder substitution (`{{var}}` via `str_replace`), path-traversal guards (slash/backslash/null byte), and an explicit `RuntimeException` for missing files.
- Adds 12 unit tests in `api/tests/Unit/Llm/SystemPromptLoaderTest.php` covering substitution, scalar coercion, missing files, traversal rejection, and a real load of both shipped prompts.

## Auto-critique

- The loader has **no dependency on `OllamaClient` (#298)** — it is a pure file-loading utility, so it can be tested and used independently. The actual prompt → LLM wiring lands in #301-#303.
- The two prompt files are **deliberately verbose** (few-shot included) to anchor the model to cycling-tour terminology. They are versioned with the code so reviews catch prompt drift.
- One PHPStan suppression (`@phpstan-ignore argument.type`) was added on the empty-string test, because the public type contract (`non-empty-string`) is what guards normal callers — the runtime check exists for defense-in-depth and the test verifies it.
- E2E tests show 16 pre-existing failures on `main` unrelated to this PR (backend-only change, no UI surface touched).

## Test plan

- [ ] CI green on the PR.
- [ ] Reviewer reads the two prompts and confirms the cycling-tour framing/terminology.

## Claude Review

**Reviewed commit:** `eb32cc5`

### Summary

The PR is in excellent shape. All three previously flagged bot-review threads have been resolved by follow-up commits: the `str_replace` sequential-substitution caveat is documented with an inline comment, `loadThrowsWhenPromptFileIsUnreadable` is implemented with a root-skip guard, and the bare `.` / `..` path-traversal gap is now closed. The `SystemPromptLoader` is a clean, well-tested, security-conscious file-loader with comprehensive edge-case coverage across 16 unit tests.

### Resolved threads
Resolved 1 previously open thread addressed by commit `eb32cc5` (bare `.` / `..` guard now added to the traversal deny-list).

### Review checklist
- [x] Code respects the project architecture
- [x] SOLID principles and Law of Demeter followed
- [x] Design patterns used where appropriate
- [x] Tests cover new/changed cases
- [x] Documentation is up to date
- [x] Dependent tickets accounted for (#298, #301–#303 referenced)

### Inline comments
No inline comments.

---
Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: api/src/Llm/SystemPromptLoader.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     *
+     * @throws \InvalidArgumentException if the prompt name is empty or contains separators
+     * @throws \RuntimeException         if the prompt file cannot be found or read
+     */
+    public function load(string $promptName, array $variables = []): string
+    {
+        if ('' === $promptName) {
+            throw new \InvalidArgumentException('Prompt name must not be empty.');
+        }
+
```
[the remark this answers is not in the document]
Comment (vincentchalamon) [reply api/src/Llm/SystemPromptLoader.php:51]: Done — added checks for `'.'` and `'..'` in the guard condition, and broadened the exception message to mention "relative-path reference". Also added `loadRejectsSingleDotPromptName` and `loadRejectsDoubleDotPromptName` tests. Pushed in fix commit eb32cc5b.
